overlord/hookstate/ctlcmd/refresh.go

```diff
@@ -0,0 +1,161 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ctlcmd
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/overlord/hookstate"
+	"github.com/snapcore/snapd/snap"
+)
+
+type refreshCommand struct {
+	baseCommand
+
+	Pending bool `long:"pending" description:"Show pending refreshes of the calling snap"`
+	// these two options are mutually exclusive
+	Proceed bool `long:"proceed" description:"Proceed with potentially disruptive refreshes"`
+	Hold    bool `long:"hold" description:"Do not proceed with potentially disruptive refreshes"`
+}
+
+var shortRefreshHelp = i18n.G("The refresh command prints pending refreshes and can hold disruptive ones.")
+var longRefreshHelp = i18n.G(`
+The refresh command prints pending refreshes of the calling snap and can hold
+disruptive refreshes of other snaps, such as refreshes of kernel or
+base snaps that can trigger restart. This command can be used from
+gate-auto-refresh hook. The hook is only run during auto-refresh.
+
+Snap can query pending refreshes with:
+    $ snapctl refresh --pending
+    pending: ready
+    channel: stable
+    version: 2
+    revision: 2
+    base: false
+    restart: false
+
+The 'pending' flag can be "ready", "none" or "inhibited". It is set to "none" if
+there are no pending refreshes for the snap. It is set to "ready" if there are,
+and to inhibited if there are but are inhibited because some of the snap
+applications are running and "refresh app awareness" feature is enabled.
```

Comment:
A few edits to make this a little more direct and less ambiguous, and also to use "held back" to link the inhibit status to the earlier description:
```
The 'pending' flag can be "ready", "none" or "inhibited". It is set to "none"
when a snap has no pending refreshes. It is set to "ready" when there are
pending refreshes and to ”inhibited” when pending refreshes are being
held back because more or more snap applications are running with the
“refresh app awareness” feature enabled.
```